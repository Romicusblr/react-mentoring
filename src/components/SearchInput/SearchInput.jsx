import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import style from './SearchInput.module.css';

function SearchInput({ placeholder, className, ...other }) {
  return (
    <input
      type="text"
      placeholder={placeholder}
      className={classNames(style.input, className)}
      {...other}
    />
  );
}

SearchInput.propTypes = {
  placeholder: PropTypes.string,
  className: PropTypes.string,
};

SearchInput.defaultProps = {
  placeholder: 'type here',
  className: '',
};

export default SearchInput;
