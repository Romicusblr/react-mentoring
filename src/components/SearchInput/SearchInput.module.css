.input {
  width: 700px;
  padding: 1rem;
  background-color: #1D1D1DCC;
  border-radius: .25rem;
  border-color: transparent;
  color: var(--color-white);
}
