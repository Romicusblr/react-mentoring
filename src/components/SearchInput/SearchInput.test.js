import SearchInput from '.';

describe('SearchInput', () => {
  it('should match to snapshot', () => {
    const snap = shallow(<SearchInput />);
    expect(snap).toMatchSnapshot();
  });
});
