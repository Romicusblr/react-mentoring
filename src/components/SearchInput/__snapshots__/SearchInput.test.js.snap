// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput should match to snapshot 1`] = `
<input
  className="input"
  placeholder="type here"
  type="text"
/>
`;
